### DISCUSSION TITLE
Can dataBinary be stored without encryption?

### DISCUSSION THREAD

**Discussion by mitht** [apache/skywalking#11643] [Q&A] [answered]
db: es7
apm version 9.3.0
agent version 8.14.0
Can dataBinary be stored without encryption? I need to group and deduplicate the db.statement inside.
I don’t want to use java code to remove duplicates. The amount of data is too large.
I want to perform deduplication processing when querying through es

I want to group "SELECT  id,name  FROM test \n \n WHERE (name = ?)"

Decrypted data:

traceId: "e0d76e7fd9f742d6947ae7e3726b4876.121.17023048191750001"
traceSegmentId: "e0d76e7fd9f742d6947ae7e3726b4876.121.17023048191750000"
spans {
  spanId: 1
  startTime: 1702304819180
  endTime: 1702304819180
  operationName: "Druid/Connection/getConnection"
  spanType: Local
  componentId: 115
}
spans {
  spanId: 2
  startTime: 1702304819180
  endTime: 1702304819201
  operationName: "PostgreSQL/JDBC/PreparedStatement/executeWithFlags"
  peer: "172.20.10.4:5432"
  spanType: Exit
  spanLayer: Database
  componentId: 37
  tags {
    key: "db.type"
    value: "PostgreSQL"
  }
  tags {
    key: "db.instance"
    value: "testdb?serverTimezone=Asia/Shanghai&useUnicode=true&characterEncoding=utf8&characterSetResults=utf8&useSSL=false&allowMultiQueries=true"
  }
  tags {
    key: "db.statement"
    value: "SELECT  id,name  FROM test \n \n WHERE (name = ?)"
  }
  tags {
    key: "db.sql.parameters"
  }
}
spans {
  spanId: 3
  startTime: 1702304819202
  endTime: 1702304819202
  operationName: "Druid/Connection/close"
  spanType: Local
  componentId: 115
}
spans {
  parentSpanId: -1
  startTime: 1702304819175
  endTime: 1702304819209
  operationName: "GET:/api/query"
  spanLayer: Http
  componentId: 14
  tags {
    key: "url"
    value: "http://127.0.0.1:8888/api/query"
  }
  tags {
    key: "http.method"
    value: "GET"
  }
  tags {
    key: "http.status_code"
    value: "200"
  }
}
service: "direct-web-01"
serviceInstance: "aadc9f99d9594d7580cf14d63571fdef@172.16.23.1"

**Comment by wu-sheng** [comment, marked as answer]
First, this is not encrypted, it is encoded only.
Two, this column is not just encoded, it is not indexing, so, no query could be applied on that.

If you want to change that, you need to change the source codes on your own, and payload and resource costs of your elasticsearch server will increase a lot.

**Reply by mitht** [reply]
thank you for your answer
